dotCMS/src/main/java/com/dotcms/publishing/job/SiteSearchJobImpl.java

```diff
@@ -128,183 +138,227 @@ public void run(final JobExecutionContext jobContext) throws JobExecutionExcepti
+                    .equals(dataMap.getString(INCLUDE_EXCLUDE)));
 
-        Logger.info(SiteSearchJobImpl.class, () -> String
-                .format(" Incremental mode [%s]. current index is `%s`. new index is `%s`. bundle id is `%s` ",
-                        BooleanUtils.toStringYesNo(incremental), indexName ,
-                        UtilMethods.isSet(newIndexName) ? newIndexName : "N/A",
-                        bundleId));
-
-        final List<Host> hosts;
-        if (indexAll) {
-            hosts = hostAPI.findAll(userToRun, true);
-        } else {
-            hosts = Stream.of(indexHosts).map(h -> {
-                try {
-                   return hostAPI.find(h, userToRun, true);
-                } catch (DotDataException | DotSecurityException e) {
-                    Logger.error(SiteSearchJobImpl.class, e);
+            String path = dataMap.getString(PATHS);
+            final List<String> paths = new ArrayList<>();
+            if (path != null) {
+                path = path.replace(',', '\r');
+                path = path.replace('\n', '\r');
+                for (String x : path.split("\r")) {
+                    if (UtilMethods.isSet(x)) {
+                        paths.add(x);
+                    }
                 }
-                return null;
-            }).filter(Objects::nonNull).collect(Collectors.toList());
-        }
+            }
+            final boolean isRunNowJob = dataMap.getBooleanFromString(RUN_NOW);
+            // Run now jobs can not get the incremental treatment.
+            final String indexAlias = dataMap.getString(INDEX_ALIAS);
+            final IndexMetaData indexMetaData = getIndexMetaData(indexAlias);
+            final String newIndexName;
+            final String indexName;
+
+            final String jobName = dataMap.getString(QUARTZ_JOB_NAME);
+            final Date startDate, endDate;
+            final List<SiteSearchAudit> recentAudits = isRunNowJob ? Collections.emptyList()
+                    : siteSearchAuditAPI.findRecentAudits(jobId, 0, 1);
+
+            final boolean incremental = (incrementalParam && !isRunNowJob && !indexMetaData
+                    .isNewIndex() && !indexMetaData.isEmpty() && !recentAudits.isEmpty());
+            //We can only run incrementally if all the above pre-requisites are met.
+            if (incremental) {
+                //Incremental mode is useful only if there's already an index previously built.
+                //Incremental mode also implies that we have to have a date range to work on.
+                //So if we have an empty index or we lack of audit data we can not run incrementally.
+                //Even if the user wants to.
+                newIndexName = null;
+                endDate = jobContext.getFireTime();
+                startDate = recentAudits.get(0).getFireDate();
+                System.out.println("Starting at " + new SimpleDateFormat("yyyyMMdd-HHmmss").format(startDate));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [System.out.println is used](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5188880)